ansible_collections/arista/avd/plugins/action/validate_and_template.py

```diff
@@ -109,29 +122,34 @@ def validate_data(self, result_messages):
         Validate data according to the schema
         """
 
-        # avd_schema.validate returns a generator, which we need to run over to perform the actual validations.
-        # The returned values are a list of validation errors
-        exceptions = list(self.avd_schema.validate(self.data))
+        # avd_schema.validate returns a generator, which we iterate through in handle_exceptions to perform the actual conversions.
+        exceptions = self.avd_schema.validate(self.data)
         validation_counter = self.handle_exceptions(exceptions, self.validation_mode)
         if validation_counter:
             result_messages.append(f"{validation_counter} errors found during schema validation of input vars.")
             return False
         return True
 
     def handle_exceptions(self, exceptions, mode):
-        arista_avd_errors = [exception for exception in exceptions if isinstance(exception, AristaAvdError)]
-        for exception in arista_avd_errors:
-            message = str.encode(f"[{self.hostname}]: {exception}", "UTF-8")
+        counter = 0
+        for exception in exceptions:
+            if not isinstance(exception, AristaAvdError):
+                continue
+
+            counter += 1
+            if mode == "quiet":
+                continue
```

Comment:
if the mode is quiet there is no need to enter this for loop

Reply:
you have to, since `exceptions` is a `Generator` which requires you to iterate through it, to actually run the code.

Reply:
we also want the counter to work even without messages. Otherwise `disabled` is the right setting.

Reply:
I am still confused by this construct of the Generator having to be ran... :)